airflow/jobs/scheduler_job.py

```diff
@@ -980,89 +952,86 @@ def _create_dag_runs(self, dag_models: Iterable[DagModel], session: Session) ->
             # are not updated.
             # We opted to check DagRun existence instead
             # of catching an Integrity error and rolling back the session i.e
-            # we need to run self._update_dag_next_dagruns if the Dag Run already exists or if we
+            # we need to set dag.next_dagrun_info if the Dag Run already exists or if we
             # create a new one. This is so that in the next Scheduling loop we try to create new runs
             # instead of falling in a loop of Integrity Error.
-            if (dag.dag_id, dag_model.next_dagrun) not in active_dagruns:
-                run = dag.create_dagrun(
+            if (dag.dag_id, dag_model.next_dagrun) not in existing_dagruns:
+                dag.create_dagrun(
                     run_type=DagRunType.SCHEDULED,
                     execution_date=dag_model.next_dagrun,
-                    start_date=timezone.utcnow(),
-                    state=State.RUNNING,
+                    state=State.QUEUED,
```

Comment:
Can be a separate PR but do you think it is worth adding `queued_dttm` similar to TaskInstances:

https://github.com/apache/airflow/blob/57dcac22137bc958c1ed9f12fa54484e13411a6f/airflow/models/taskinstance.py#L307

This can help in future for benchmarking difference between Queued and Start time for a DagRun

Reply:
Yeah. That's cool, separate PR is cool

Reply:
Actually to avoid a _second_ migration, I think it makes sense to do in this PR, sorry.

(Migrations are relatively slow, so if we are already altering a table it's better to do all the changes at once.)